vslib/SwitchState.cpp

```diff
@@ -199,6 +199,102 @@ void SwitchState::asyncOnLinkMsg(
     m_switchConfig->m_eventQueue->enqueue(std::make_shared<Event>(EVENT_TYPE_NET_LINK_MSG, payload));
 }
 
+sai_status_t SwitchState::getNetStat(
+        _In_ sai_stat_id_t counter_id, 
+        _In_ std::string& if_name, 
+        _Out_ uint64_t &counter)
+{
+    struct {
+        sai_stat_id_t id;
+        std::string name;
+    } counter_map [] = {
+        { SAI_PORT_STAT_IF_IN_OCTETS, "rx_bytes" },
+        { SAI_PORT_STAT_IF_IN_UCAST_PKTS, "rx_packets" },
+        { SAI_PORT_STAT_IF_IN_ERRORS, "rx_errors" },
+        { SAI_PORT_STAT_IF_IN_DISCARDS, "rx_dropped" },
+        { SAI_PORT_STAT_IF_OUT_OCTETS, "tx_bytes" },
+        { SAI_PORT_STAT_IF_OUT_UCAST_PKTS, "tx_packets" },
+        { SAI_PORT_STAT_IF_OUT_ERRORS, "tx_errors" },
+        { SAI_PORT_STAT_IF_OUT_DISCARDS, "tx_dropped" }
+    };
+    uint32_t counter_map_size = 8;
```

Comment:
this should be obtained automatically, if you declare this entries as std::vector<strict>

Reply:
replaced with std::map no size

Reply:
Let me know how to make map const? for now i made it just private initialized in constructor. Do you have an example for const map?

Reply:
try:
static const std::map<x,y> m_map;
and it must be initialized in cpp file take a look here: https://stackoverflow.com/questions/2636303/how-to-initialize-a-private-static-const-map-in-c 
const std::map<x,y>

Reply:
ok. thanks